app/models/queries/shop_monthly_enrollments.rb

```diff
@@ -199,16 +223,16 @@ def project_enrollment_ids
 
     def collect_benefit_group_ids(effective_on = nil)
       @feins.collect{|e| prepend_zeros(e.to_s, 9) }.inject([]) do |id_list, fein|
-        employer = EmployerProfile.find_by_fein(fein)
-        if employer.present?
-          plan_years = employer.plan_years.where(:aasm_state.in => PlanYear::PUBLISHED + PlanYear::RENEWING_PUBLISHED_STATE + ['expired'])
-          plan_year = plan_years.where(:start_on => effective_on || @effective_on).first
+        benefit_sponsorship = BenefitSponsors::Organizations::Organization.where(fein:  fein).first.active_benefit_sponsorship
+
+        if benefit_sponsorship.present?
+          benefit_application = benefit_sponsorship.benefit_applications.where(:"effective_period.min" => effective_on || @effective_on , :aasm_state => :active).first
         end
 
-        if plan_year.blank? || plan_year.external_plan_year?
+        if benefit_application.blank? || benefit_application.imported?
```

Comment:
benefit_application.imported? is a redundant check, not needed...:imported is a state on the new model.